contracts/transfer/src/state.rs

```diff
@@ -40,63 +65,134 @@ impl TransferState {
             tree: Tree::new(),
             nullifiers: BTreeSet::new(),
             roots: ConstGenericRingBuffer::new(),
-            balances: BTreeMap::new(),
-            deposit: None,
+            accounts: BTreeMap::new(),
+            contract_balances: BTreeMap::new(),
+        }
+    }
+
+    fn handle_withdraw<F: FnOnce(ContractId) -> Sender>(
+        &mut self,
+        withdraw: Withdraw,
+        produce_sender: F,
```

Comment:
This can simply by the `ContractId`? Since in all use cases you are calling the same function anyway, right?

Reply:
A contract ID and a name, but yes